src/Compilers/CSharp/Portable/CodeGen/EmitStackAllocInitializer.cs

```diff
@@ -45,7 +45,11 @@ private void EmitStackAlloc(TypeSymbol type, BoundArrayInitialization? inits, Bo
                 var sizeInBytes = elementType.EnumUnderlyingTypeOrSelf().SpecialType.SizeInBytes();
 
                 ImmutableArray<byte> data = GetRawData(initExprs);
+#if NET8_0_OR_GREATER
+                if (!data.AsSpan().ContainsAnyExcept(data[0]))
```

Comment:
We don't generally have a problem with conditional code, but we usually want to have a good reason to have it; do you have benchmarks or other data indicating this capture is a problem?